### PR TITLE
Process collections

### PR DESCRIPTION
**Pull request by qwtel**
This PR adds support for #6. It also adds an option to disable the plugin to be consistent with benbalter/jekyll-optional-front-matter#9

It includes updated documentation and tests.

### CODE REVIEW

**Comment by qwtel** [comment]
Didn't see #27 before submitting this 😣

File: lib/jekyll-relative-links/generator.rb [hunk 1]
```diff
@@ -23,15 +26,19 @@ def initialize(site)
     def generate(site)
       @site    = site
       @context = context
+      return if disabled?
 
-      site.pages.each do |page|
+      pages = site.pages
+      pages = site.pages + site.documents if collections?
```
Comment (benbalter) [inline lib/jekyll-relative-links/generator.rb]: Should `documents` really be `docs_to_write`?

File: lib/jekyll-relative-links/generator.rb [hunk 2]
```diff
@@ -78,7 +85,9 @@ def url_for_path(path)
     end
 
     def potential_targets
-      @potential_targets ||= (site.pages + site.static_files)
+      @potential_targets ||= begin
+        site.pages + site.static_files + (collections? ? site.documents : [])
```
Comment (benbalter) [inline lib/jekyll-relative-links/generator.rb]: Should you be able to link *to* a collection document regardless of if collections are enabled?

File: lib/jekyll-relative-links/generator.rb [hunk 3]
```diff
@@ -23,15 +26,19 @@ def initialize(site)
...
+
+      pages.each do |page|
         next unless markdown_extension?(page.extname)
         replace_relative_links!(page)
       end
     end
 
     def replace_relative_links!(page)
-      url_base = File.dirname(page.path)
+      url_base = File.dirname(page.relative_path)
```
Comment (benbalter) [inline lib/jekyll-relative-links/generator.rb]: I ran into some strange behavior with this. https://github.com/benbalter/jekyll-relative-links/pull/27/files#diff-1c96af68dcbaa3116d54508be6c5ca8bR140 was my work around to deal with the different behavior between pages and documents when it comes to paths.

File: lib/jekyll-relative-links/generator.rb (hunk 1 above)
Comment (qwtel) [reply lib/jekyll-relative-links/generator.rb]: yes it should! I wasn't aware of this property until just now, but it does exactly what we want.

File: lib/jekyll-relative-links/generator.rb (hunk 3 above)
Comment (qwtel) [reply lib/jekyll-relative-links/generator.rb]: hm, on the surface it looks like it `path` and `relative_path` are always identical for pages. do have more details on where this behavior occurred?

File: lib/jekyll-relative-links/generator.rb (hunk 2 above)
Comment (qwtel) [reply lib/jekyll-relative-links/generator.rb]: hm, coming to think about it, the collections option could be enabled by default for this plugin. Are there meaningful ways to use relative links inside a post/collection that would be broken by this change?

Comment (qwtel) [reply lib/jekyll-relative-links/generator.rb]: in any case, `site.docs_to_write` can be part of the potential targets.

**Comment by qwtel** [comment]
Glad you like it. I didn't look into the other solution in detail, I only noticed that it is more code. This could mean I'm missing edge-cases though. I'll leave the reviewing to you :wink:

I thought about per-collection support, but wanted to see if anybody even needs it (more likely the case with `titles-from-headings`'s collection support I think). Anyway, the way to extend the config would be to accept arrays

```yaml
relative_links:
  collections: [posts, items]
```

File: README.md [hunk 4]
```diff
@@ -48,3 +48,42 @@ Because Markdown files rendered by GitHub Pages should behave similar to Markdow
...
+
+~~~
+├── _posts
+│   └── 2016-01-01-test.md
+├── _config.yml
+├── index.md
+└── some-category
+    └── _posts
+        └── 2016-01-01-test.md
+~~~
```
Comment (janpio) [inline README.md]: Maybe add a `_my_collection` folder with a `some_doc.md` file to make these examples match the official collections documentation from https://jekyllrb.com/docs/collections/ here?

(At first I was confused if this is some special kind of _posts and not a real collection how I know them)

File: README.md [hunk 5]
```diff
@@ -48,3 +48,42 @@ Because Markdown files rendered by GitHub Pages should behave similar to Markdow
...
     - jekyll-relative-links
   ```
+
+## Configuration
+
+You can configure this plugin in `_config.yml` under the `relative_links` key. This is optional and defaults to:
+
+```yml
+relative_links:
+  enabled:     true
```
Comment (janpio) [inline README.md]: Technically disabling/enabling the plugin is out of scope of the issue being solved, or is this needed for this functionality?

Comment (janpio) [reply README.md]: (But as this is super useful, I of course would have no problem this making its way into the code)
(Review mechanism might make it look otherwise, was not my intention - should have just commented the code)

**Comment by janpio** [comment]
I am really looking forward for this being merged. Will make my editing process so much easier. 
Also super readable code and simple changes.

File: README.md (hunk 5 above)
Comment (qwtel) [reply README.md]: yes thats correct. it's a separate commit though

**Comment by benbalter** [comment]
@qwtel 👍 to your suggestion regarding per-collection configuration. Easy enough to add support for more finer grained controls after the fact if users want it.

File: spec/fixtures/site/page.md [hunk 6]
```diff
@@ -27,8 +27,12 @@
...
 
+[A indented reference][indented-reference]
+
 [reference]: another-page.md
 
 [reference-with-fragment]: another-page.md#foo
 
 [reference-brackets]: another-page.md#(bar)
+
+  [indented-reference]: another-page.md
```
Comment (benbalter) [inline spec/fixtures/site/page.md]: Just to ensure we are testing every edge case and further work doesn't break this, perhaps we should test linking from a page to a document?

File: lib/jekyll-relative-links/generator.rb (hunk 3 above)
Comment (benbalter) [reply lib/jekyll-relative-links/generator.rb]: Read through the code and the above linked PR and I think I figured out the issue:

https://github.com/benbalter/jekyll-relative-links/blob/36318fd8699013a538823c86b8da9340c84dad77/lib/jekyll-relative-links/generator.rb#L34 is intended to set `url_base` as the path to the linking file's parent directory relative to the site source. That's then passed to [`path_from_root`](https://github.com/benbalter/jekyll-relative-links/blob/36318fd8699013a538823c86b8da9340c84dad77/lib/jekyll-relative-links/generator.rb#L40) as the URL base along with the relative path to the link to determine the linked file's path relative to the site source.

I think the issue here may be that Pages and Documents would behave differently when the site's source was not the working directory (or you may not be able to link from a page to a document) since the paths would not match.

File: spec/fixtures/site/page.md (hunk 6 above)
Comment (qwtel) [reply spec/fixtures/site/page.md]: oops, totally forgot adding tests for those.

**Comment by benbalter** [comment]
Thanks again @qwtel!

**Comment by janpio** [comment]
🥇 

Can't wait for this to get released and included on Github Pages!
